scipy/stats/tests/test_mstats_basic.py

```diff
@@ -223,6 +231,13 @@ def test_kendalltau(self):
                             25,80,80,80,80,80,80, 0,10,45, np.nan, 0])
         result = mstats.kendalltau(x,y)
         assert_almost_equal(np.asarray(result), [-0.1585188, 0.4128009])
+        # make sure internal variable use correct precision with
+        # larger arrays
+        x = np.arange(2000, dtype=np.float)
+        x = ma.masked_greater(x, 1995)
+        y = np.arange(2000, dtype=np.float)
+        y = np.concatenate((y[1000:], y[:1000]))
+        assert_almost_equal(mstats.kendalltau(x,y)[1], 0.97344095)
```

Comment:
Well, if the test is testing that rho is not nan, then `assert_(np.isfinite(...))` is an option. Of course, if you can do better, great. Then I guess it'd be nice to have a comparison with R's `cor.test`.